core/src/test/scala/org/apache/spark/SparkFunSuite.scala

```diff
@@ -374,6 +382,19 @@ abstract class SparkFunSuite
     checkError(exception, errorClass, sqlState, parameters,
       matchPVals = true, Array(context))
 
+  protected def checkErrorTableNotFound(
```

Comment:
Here is the core, right? which is not aware of any tables. Please, move it from here.

Reply:
@MaxGekk OK, where can I put it instead? This little function eliminates dozens of more complicated checkError invocations across many test files.
I added it because it's so pervasive

Reply:
How about `QueryTest`, and pass a SQL statement or a plan into `checkErrorTableNotFound()`.

Reply:
There are two patterns:
`      val e = intercept[AnalysisException] {
        sql(s"ALTER TABLE $catalog.ns.no_tbl PARTITION (id=1) RENAME TO PARTITION (id=2)")
      }
      checkErrorTableNotFound(e, s"`$catalog`.`ns`.`no_tbl`",
        ExpectedContext(s"$catalog.ns.no_tbl", 12, 11 + s"$catalog.ns.no_tbl".length))
`
and
`
    val exc = intercept[NoSuchTableException] {
      catalog.loadTable(testIdent)
    }

    checkErrorTableNotFound(exc, testIdentQuoted)
`

Not sure how the API of that new shared function would look like.
@MaxGekk @cloud-fan